test/integration/fixtures/server-through-relay.js

```diff
@@ -0,0 +1,37 @@
+const DHT = require('../../../')
+
+const publicKey = Buffer.from(process.argv[2], 'hex')
+const secretKey = Buffer.from(process.argv[3], 'hex')
+const relayServer = Buffer.from(process.argv[4], 'hex')
+const socketKeepAlive = Number(process.argv[5] || 5000)
+const relayKeepAlive = Number(process.argv[6] || 5000)
+const keyPair = { publicKey, secretKey }
+
+main()
+
+async function main () {
+  const node = new DHT()
+  const server = node.createServer({
+    holepunch: false, // To ensure it relies only on relaying
+    shareLocalAddress: false, // To help ensure it relies only on relaying (otherwise it can connect directly over LAN, without even trying to holepunch)
+    relayKeepAlive,
+    relayThrough: relayServer
+  }, socket => {
+    socket.setKeepAlive(socketKeepAlive)
+    socket
+      .on('data', data => {
+        console.log(`socket_ondata ${data.toString()}`)
+        socket.write('world')
+      })
+      .on('open', () => console.log(`socket_onopen ${socket.rawStream.remoteHost}:${socket.rawStream.remotePort}`))
+      .on('close', () => console.log('socket_onclose'))
+      .on('error', err => console.log(`socket_onerror ${err.code}`))
```

Comment:
Detail: I don't think `err` is guaranteed to have a `code` prop, so `err?.code` seems safer (same for the other err handler)

Reply:
Edit: it's actually fine, since it will just print null, so nvm